frontend/packages/operator-lifecycle-manager/src/components/modals/uninstall-operator-modal.tsx

```diff
@@ -32,26 +47,42 @@ export const UninstallOperatorModal: React.FC<UninstallOperatorModalProps> = ({
         apiVersion: 'v1',
         propagationPolicy: 'Foreground',
       };
-      handlePromise(
-        Promise.all([
-          k8sKill(SubscriptionModel, subscription, {}, deleteOptions),
-          ...(subscription?.status?.installedCSV
-            ? [
-                k8sKill(
-                  ClusterServiceVersionModel,
-                  {
-                    metadata: {
-                      name: subscription.status.installedCSV,
-                      namespace: subscription.metadata.namespace,
-                    },
+      const k8kills = [
+        k8sKill(SubscriptionModel, subscription, {}, deleteOptions),
+        ...(subscription?.status?.installedCSV
+          ? [
+              k8sKill(
+                ClusterServiceVersionModel,
+                {
+                  metadata: {
+                    name: subscription.status.installedCSV,
+                    namespace: subscription.metadata.namespace,
                   },
-                  {},
-                  deleteOptions,
-                ).catch(() => Promise.resolve()),
-              ]
-            : []),
-        ]),
-      )
+                },
+                {},
+                deleteOptions,
+              ).catch(() => Promise.resolve()),
+            ]
+          : []),
+      ];
+      if (deleteAllOperands && operands) {
+        operands.forEach((operand: K8sResourceKind) => {
+          const model =
+            models?.[
+              operand.apiVersion
+                .split('/')
+                .join('~')
+                .concat(`~${operand.kind}`)
+            ];
+          if (model) {
+            k8kills.push(k8sKill(model, operand, {}, deleteOptions));
```

Comment:
We need error handling.

Reply:
I have improved error handling for the core 'Uninstall Operator' as well as "Delete All Operands" operations.   I met with UXD and they have approved (with minor tweaks) the use of a 'submit results' view after the Uninstall (submit) button is clicked in the uninstall modal.

Please see these [screenshots](https://docs.google.com/document/d/1iwx7cn2RICfYBf80YEJ0BFjG64q2xxBu6jbXLgP6kpc/edit#)!